diff -u -r -U5 feh-2.1/config.mk feh-2.1.docdir/config.mk
--- feh-2.1/config.mk	2011-11-02 05:59:20.000000000 -0400
+++ feh-2.1.docdir/config.mk	2011-11-22 22:09:30.243996677 -0500
@@ -6,11 +6,11 @@
 
 # Directories for manuals, executables, docs, data, etc.
 main_dir = ${DESTDIR}${PREFIX}
 man_dir = ${main_dir}/share/man
 bin_dir = ${main_dir}/bin
-doc_dir = ${main_dir}/share/doc/feh
+doc_dir = ${main_dir}/share/doc/feh-${VERSION}
 image_dir = ${main_dir}/share/feh/images
 font_dir = ${main_dir}/share/feh/fonts
 example_dir = ${main_dir}/share/doc/feh/examples
 
 # default CFLAGS
